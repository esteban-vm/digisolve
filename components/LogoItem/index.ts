export { default as LogoItem, type LogoProps } from './LogoItem'
