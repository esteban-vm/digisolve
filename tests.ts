import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
