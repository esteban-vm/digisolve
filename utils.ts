export const formatPrice = (price: number) => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 0,
  }).format(price)
}
