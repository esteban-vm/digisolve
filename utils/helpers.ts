import Filter from 'bad-words'

const filter = new Filter()
/** Checks if a message does not contain profane language */
export const isClean = (message: string) => !filter.isProfane(message)

const formatter = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD', maximumFractionDigits: 0 })
/** Formats a price in number to USD currency */
export const formatPrice = (price: number) => formatter.format(price)

/** Gets the name of an icon from kebab case to pascal case */
export const getIconName = <T extends string>(name: T) => {
  return name.replace(/(^\w|-\w)/g, (name) => name.replace(/-/, '').toUpperCase())
}
