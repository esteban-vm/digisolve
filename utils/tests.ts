import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
