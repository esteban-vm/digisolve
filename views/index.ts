export * from './About'
export * from './Header'
export * from './Partners'
export * from './Portfolio'
export * from './Services'
export * from './Testimonials'
