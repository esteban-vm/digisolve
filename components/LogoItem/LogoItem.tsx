import type { FunctionalComponent } from '@/types'
import Image, { type StaticImageData } from 'next/image'
import { styled } from '@/utils/styles'

/** Properties for logo item */
export type LogoProps = {
  /** The logo image */
  logo: StaticImageData
  /** The alt text */
  alt: string
}

const LogoItemComponent: FunctionalComponent<LogoProps> = ({ logo, alt, ...rest }) => {
  return (
    <div {...rest}>
      <Image src={logo} alt={alt} />
    </div>
  )
}

const LogoItem = styled(LogoItemComponent)`
  img {
    width: 65%;
    height: auto;
  }
`

LogoItem.defaultProps = {
  className: 'col span_1_of_4',
}

export default LogoItem
