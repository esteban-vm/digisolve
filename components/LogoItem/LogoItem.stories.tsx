import type { Meta, StoryObj } from '@storybook/react'
import { within, userEvent } from '@storybook/testing-library'
import { styled } from '@/utils/styles'
import { logos } from '@/views'
import LogoItem from './LogoItem'

const Wrapper = styled.div`
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 75vw;

  div {
    background-color: var(--color-light);
  }
`

export default {
  argTypes: {
    logo: {
      control: false,
    },
    alt: {
      control: false,
    },
    onMouseEnter: {
      action: true,
      control: false,
    },
  },
  component: LogoItem,
  title: 'Components/LogoItem',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const logo = canvas.getByRole('img')
    userEvent.hover(logo)
  },
} as Meta<typeof LogoItem>

const [logo1, logo2, logo3, logo4] = logos

export const LogoExample1: StoryObj<typeof LogoItem> = { args: logo1 }
export const LogoExample2: StoryObj<typeof LogoItem> = { args: logo2 }
export const LogoExample3: StoryObj<typeof LogoItem> = { args: logo3 }
export const LogoExample4: StoryObj<typeof LogoItem> = { args: logo4 }
