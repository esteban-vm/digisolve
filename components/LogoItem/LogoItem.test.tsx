import { render, cleanup, create, screen } from '@/utils/tests'
import { logos } from '@/views'
import LogoItem from './LogoItem'

describe('ðŸ§ª <LogoItem /> test cases:', () => {
  const logo = <LogoItem {...logos[0]} />

  describe('there should be:', () => {
    beforeEach(() => {
      render(logo)
    })

    afterEach(cleanup)

    it('a logo', () => {
      const logoImg = screen.getByRole('img')
      expect(logoImg).toBeInTheDocument()
      expect(logoImg).toBeVisible()
    })

    it('an accessible text', () => {
      const logoImg = screen.getByRole('img')
      expect(logoImg).toHaveAccessibleName(/^brand logo 1$/i)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const logoTree = create(logo).toJSON()
      expect(logoTree).toMatchSnapshot()
    })
  })
})
