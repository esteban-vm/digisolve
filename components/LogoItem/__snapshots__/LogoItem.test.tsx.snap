// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª <LogoItem /> test cases: style tests: should render with correct styles 1`] = `
.emotion-0 img {
  width: 65%;
  height: auto;
}

<div
  className="col span_1_of_4 emotion-0"
>
  <img
    alt="Brand Logo 1"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={40}
  />
</div>
`;
